### ISSUE TITLE
Is it a way to get the percentage?

### ISSUE THREAD

**Issue by mayuanyang** [Microsoft/CNTK#886]
I use the C# wrapper for image class evaluation, i currently has only 2 class "Lion" and "Horse", the output doesn't look like a percentage number, is it a way to get it? e.g. i am expecting a result of 
Index   Value
0          89.1%
1          10.9%

thanks

**Comment by frankseide** [comment]
What is your output node? In many examples, we do not compute the Softmax, but just emit the argument to the Softmax, since this is sufficient for classification.

You can either add a Softmax node to your network, or manually convert the values you get in your code by implementing Softmax.

**Comment by mayuanyang** [comment]
Not too sure if i understand your question correctly, i got 
`ce       = CrossEntropyWithSoftmax (labels, z)`
Is it what you are talking about? how can i get it from the C# Eval?

**Comment by frankseide** [comment]
I think your output is `z`. If you want Softmax values, please add this:

    P = Softmax (z)

and change the `outputNodes` from `z` to `P`.

**Comment by mayuanyang** [comment]
Cool, that works. I do have another question, as i only got two classes "Horse" and "Lion", when i upload an image is neither a horse nor lion, it still return me a value of e.g. 70% for horse and 30% for lion, i understand that it is because of the Softmax, but can i add a thrid class "Other" e.g.
Horse: 1%
Lion: 1%
Other: 98%

thanks a lot

**Comment by frankseide** [comment]
You'd have to train it that way. If you have a lot of classes (thousands), you could hope that an unknown category would give you multiple classes that respond, weighting each other down. But with only 2 classes, I think you would have to explicitly train a "none-of-the-above" model.

**Comment by mayuanyang** [comment]
Sounds like a good idea, by training a "none-of-the-above" model, what sort of images should i use? I have trained my model with additional label but doesn't help

**Comment by cha-zhang** [comment]
You need all kind of images that are not lion or horse, preferably a very large quantity. This is not trivial since most likely now your data would be heavily biased. 

There are some research papers trying to address none-of-the-above issue, such as this one: 
https://arxiv.org/abs/1511.06233

**Comment by mayuanyang** [comment]
Cool, thanks for the info, so does CNTK provide something to OpenMax?

**Comment by zhouwangzw** [comment]
@mayuanyang regarding your question "How can I get it from the C# Eval", you just need to put the node name into the output, e.g.
```
outputs = new Dictionary<string, List<float>>()
                    {   { "out.P", you_output_array }  };
model.Evaluate(inputs, outputs);
```

**Comment by mayuanyang** [comment]
Thanks a lot for your help
